examples/www-image/main/jpg-render.c

```diff
@@ -0,0 +1,518 @@
+        .pmf_cfg = {
+            .capable = true,
+            .required = false
+        },
+    },
+    };
+
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA));
+    ESP_ERROR_CHECK(esp_wifi_set_config((wifi_interface_t)ESP_IF_WIFI_STA, &wifi_config));
+    ESP_ERROR_CHECK(esp_wifi_start());
+    ESP_LOGI(TAG, "wifi_init_sta finished.");
+
+    /* Waiting until either the connection is established (WIFI_CONNECTED_BIT) or connection failed for the maximum
+     * number of re-tries (WIFI_FAIL_BIT). The bits are set by event_handler() (see above) */
+    EventBits_t bits = xEventGroupWaitBits(s_wifi_event_group,
+                                           WIFI_CONNECTED_BIT | WIFI_FAIL_BIT,
+                                           pdFALSE,
+                                           pdFALSE,
+                                           portMAX_DELAY);
+
+    /* xEventGroupWaitBits() returns the bits before the call returned, hence we can test which event actually happened. */
+    if (bits & WIFI_CONNECTED_BIT)
+    {
+        ESP_LOGI(TAG, "Connected to ap SSID:%s", ESP_WIFI_SSID);
+    }
+    else if (bits & WIFI_FAIL_BIT)
+    {
+        ESP_LOGI(TAG, "Failed to connect to SSID:%s", ESP_WIFI_SSID);
+    }
+    else
+    {
+        ESP_LOGE(TAG, "UNEXPECTED EVENT");
+    }
+
+    /* The event will not be processed after unregister */
+    ESP_ERROR_CHECK(esp_event_handler_instance_unregister(IP_EVENT, IP_EVENT_STA_GOT_IP, instance_got_ip));
+    ESP_ERROR_CHECK(esp_event_handler_instance_unregister(WIFI_EVENT, ESP_EVENT_ANY_ID, instance_any_id));
+    vEventGroupDelete(s_wifi_event_group);
+}
+
+void app_main() {
+  epd_init(EPD_OPTIONS_DEFAULT);
+  hl = epd_hl_init(WAVEFORM);
+  fb = epd_hl_get_framebuffer(&hl);
+
+  decoded_image = (uint8_t *)heap_caps_malloc(EPD_WIDTH * EPD_HEIGHT, MALLOC_CAP_SPIRAM);
+  if (decoded_image == NULL) {
+      ESP_LOGE("main", "Initial alloc back_buf failed!");
+  }
+  memset(decoded_image, 255, EPD_WIDTH * EPD_HEIGHT);
+
+  // Should be big enough to allocate the JPEG file size
+  source_buf = (uint8_t *)heap_caps_malloc(EPD_WIDTH * EPD_HEIGHT, MALLOC_CAP_SPIRAM);
+  if (source_buf == NULL) {
+      ESP_LOGE("main", "Initial alloc source_buf failed!");
+  }
+  printf("Free heap after buffers allocation: %d\n", xPortGetFreeHeapSize());
+
+  double gammaCorrection = 1.0 / gamma_value;
+  for (int gray_value =0; gray_value<256;gray_value++)
+    gamme_curve[gray_value]= round (255*pow(gray_value/255.0, gammaCorrection));
+
+  // Initialize NVS
+  esp_err_t ret = nvs_flash_init();
+  if (ret == ESP_ERR_NVS_NO_FREE_PAGES || ret == ESP_ERR_NVS_NEW_VERSION_FOUND)
+  {
+      ESP_ERROR_CHECK(nvs_flash_erase());
+      ret = nvs_flash_init();
+  }
+  ESP_ERROR_CHECK(ret);
+
+  // WiFi log level set only to Error otherwise outputs too much
+  esp_log_level_set("wifi", ESP_LOG_ERROR);
+  
+  // Initialization: WiFi + clean screen while downloading (optional)
+  wifi_init_sta();
+  epd_poweron();
```

Comment:
The display should be cleared, because the highlevel API will forget the screen state after deepsleep (The PSRAM is powered down). To get accurate grayscale, the screen must be brought to a known state. If using the highlevel API, use `epd_fullclear(&hl, temp)`.

Also, we only powering on before draw (and clear) will waste less power.

Reply:
Added thanks!